src/main/src/main/java/org/geoserver/catalog/ResourcePoolCallback.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *  (c) 2019 Open Source Geospatial Foundation - all rights reserved
+ *  This code is licensed under the GPL 2.0 license, available at the root
+ *  application directory.
+ */
+
+package org.geoserver.catalog;
+
+import org.geotools.data.FeatureSource;
+import org.opengis.feature.Feature;
+import org.opengis.feature.type.FeatureType;
+
+import java.io.Serializable;
+
+/**
+ * Basic Extension for {@link ResourcePool} class. Intended for rebuilding {@link FeatureType} and
+ * getting modified features through wrapper {@link FeatureSource}
+ *
+ * @param <FT> feature type class
+ * @param <F> feature class
+ */
+public interface ResourcePoolCallback<FT extends FeatureType, F extends Feature>
```

Comment:
Fixed

Reply:
Some more info in this one, the extension point interface was simplified and renamed too ``RetypeFeatureTypeCallback``, because well what it allow us to do is to retype a feature type, and consequently the associated features by wrapping the feature source.

The existing ``FeatureTypeCallBack`` as a different propose, it allows an extension to create a non existing feature type a in a certain data source, which means that the data source needs to eb aware of this, e.g. SQL views or Solr types. 

Which means that this extension points \ call backs are also invoked in different contexts, in fact, the ``RetypeFeatureTypeCallback`` should be (and it is) to retype feature types created by ``FeatureTypeCallback``, e.g. SQL views.

That say, let us know your judgement on this new name @aaime :bowtie:.

Reply:
That's a good name indeed.